File: InPlan.Web/Modules/Parametros/AmbientesProyectos/AmbientesProyectosForm.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace InPlan.Parametros.Forms;

[FormScript("Parametros.AmbientesProyectos")]
[BasedOnRow(typeof(AmbientesProyectosRow), CheckNames = true)]
public class AmbientesProyectosForm
{
    [Tab("Datos")]
    [HalfWidth]
    public int IdProyecto { get; set; }
    [HalfWidth]
    public int IdAmbiente { get; set; }
    public string ServidorWeb { get; set; }
    public string InstaciaServidor { get; set; }
    [HalfWidth, TextAreaEditor(Rows = 4)]
    public string ServidorDatos { get; set; }
    [HalfWidth, TextAreaEditor(Rows = 4)]
    public string NotasAdicionales { get; set; }
    [TextAreaEditor(Rows = 5)]
    public string Observaciones { get; set; }
    [Tab("Documentos")]
    public string DocumentosRelacionados { get; set; }
}

File: InPlan.Web/Modules/ServerTypes/Parametros/AmbientesProyectosForm.ts
import { LookupEditor, StringEditor, TextAreaEditor, MultipleImageUploadEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface AmbientesProyectosForm {
    IdProyecto: LookupEditor;
    IdAmbiente: LookupEditor;
    ServidorWeb: StringEditor;
    InstaciaServidor: StringEditor;
    ServidorDatos: TextAreaEditor;
    NotasAdicionales: TextAreaEditor;
    Observaciones: TextAreaEditor;
    DocumentosRelacionados: MultipleImageUploadEditor;
}

export class AmbientesProyectosForm extends PrefixedContext {
    static formKey = 'Parametros.AmbientesProyectos';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!AmbientesProyectosForm.init)  {
            AmbientesProyectosForm.init = true;

            var w0 = LookupEditor;
            var w1 = StringEditor;
            var w2 = TextAreaEditor;
            var w3 = MultipleImageUploadEditor;

            initFormType(AmbientesProyectosForm, [
                'IdProyecto', w0,
                'IdAmbiente', w0,
                'ServidorWeb', w1,
                'InstaciaServidor', w1,
                'ServidorDatos', w2,
                'NotasAdicionales', w2,
                'Observaciones', w2,
                'DocumentosRelacionados', w3
            ]);
        }
    }
}